pyproject.toml

```diff
@@ -23,3 +23,25 @@ markers = [
 [tool.black]
 target-version = ['py38', 'py39', 'py310']
 line-length = 88
+
+# https://github.com/charliermarsh/ruff
+[tool.ruff]
+line-length = 88
+target-version = "py38"
+src = ["npe2", "tests"]
+select = [
+    "E", "F", "W", #flake8
+    "UP", # pyupgrade
+    "I", # isort
+    "B", # flake8-bugbear
+    "C4", # flake8-comprehensions
+    "TID",  # flake8-tidy-imports
+    "RUF",  # ruff-specific rules
+]
+
+[tool.ruff.per-file-ignores]
+"npe2/cli.py" = ["B008","A00"]
+
+[tool.ruff.pyupgrade]
+# Preserve types, even if a file imports `from __future__ import annotations`.
+keep-runtime-typing = true
```

Comment:
```suggestion
[tool.ruff.pyupgrade]
# Preserve types, even if a file imports `from __future__ import annotations`.
keep-runtime-typing = true

[tool.ruff.isort]
known-first-party=['npe2']
```

For proper separation of `npe2` import

Reply:
(just gave all core devs write permission)